pkg/tls/tls.go

```diff
@@ -134,50 +159,61 @@ func GetClientTLSConfig(clientLog *zerolog.Logger, client kclient.Client) (*tls.
 	}
-
-	caCertPool := x509.NewCertPool()
-	for key, element := range CACertsData {
-		// skip non cerificate keys like crt.key if exists in the secret
-		if !strings.HasSuffix(key, TLSCertKeySuffix) {
-			continue
+	if !cacertProvided {
+		// no private CA certificates was provided.
+		clientLog.Info().Msg("no private CA certificates were provided")
+	} else {
+		CACertsData, e := GetCertificatesFromSecret(client, caSecretName, caSecretNamespace)
+		if e != nil {
+			clientLog.Error().Err(e).Msg("error in GetCertificatesFromSecret() tring to get CA cert")
+			return nil, e
 		}
-		if !caCertPool.AppendCertsFromPEM(element) {
-			clientLog.Error().Err(err).Msg("error in AppendCertsFromPEM trying to load: " + key)
-			return nil, errors.New("error in GetClientTLSConfig in AppendCertsFromPEMtrying to load: " + key)
+		clientLog.Info().Msg("private CA certificates are provided")
+		caCertPool, err = x509.SystemCertPool()
+		if err != nil {
+			return nil, err
 		}
-	}
-
-	var tlsConfig *tls.Config
-	if certSecretName == "" || certSecretNamespace == "" {
-		clientLog.Info().Msg(MTLSDisabledMsg)
-		tlsConfig = &tls.Config{
-			RootCAs:    caCertPool,
-			MinVersion: tls.VersionTLS13,
+		for key, element := range CACertsData {
+			// skip private key
+			if key == corev1.TLSPrivateKeyKey {
+				continue
+			}
+			if !caCertPool.AppendCertsFromPEM(element) {
+				clientLog.Error().Err(err).Msg("error in AppendCertsFromPEM trying to load: " + key)
+				return nil, errors.New("error in GetClientTLSConfig in AppendCertsFromPEMtrying to load: " + key)
+			}
 		}
-		return tlsConfig, nil
 	}
 
-	clientLog.Info().Msg(MTLSEnabledMsg)
-	clientCertsData, err := GetCertificatesFromSecret(client, certSecretName, certSecretNamespace)
+	var tlsConfig *tls.Config
+	certProvided, err := isCertificateProvided(certSecretName, certSecretNamespace)
 	if err != nil {
-		clientLog.Error().Err(err).Msg("error in GetCertificatesFromSecret tring to get client/server cert")
+		clientLog.Error().Err(err).Msg("error in the provided client certificate")
 		return nil, err
 	}
-	cert, err := tls.X509KeyPair(clientCertsData[corev1.TLSCertKey], clientCertsData[corev1.TLSPrivateKeyKey])
-	if err != nil {
-		clientLog.Error().Err(err).Msg("error in X509KeyPair")
-		return nil, err
+	var cert tls.Certificate
+	if certProvided {
+		clientLog.Log().Msg("TLS certificates were provided for the connector")
+
+		clientCertsData, err := GetCertificatesFromSecret(client, certSecretName, certSecretNamespace)
+		if err != nil {
+			clientLog.Error().Err(err).Msg("error in GetCertificatesFromSecret tring to get client/server cert")
+			return nil, err
+		}
+		cert, err = tls.X509KeyPair(clientCertsData[corev1.TLSCertKey], clientCertsData[corev1.TLSPrivateKeyKey])
+		if err != nil {
+			clientLog.Error().Err(err).Msg("error in X509KeyPair")
+			return nil, err
+		}
+	} else if !cacertProvided {
+		clientLog.Log().Msg("no TLS certificates were provided for the client")
+		return nil, nil
```

Comment:
Line 104, 142, 221:, MinVersion:   tls.VersionTLS13 should be configurable both for client and server

Reply:
Changed. Thanks.